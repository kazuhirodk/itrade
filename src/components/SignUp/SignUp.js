import React from 'react';
import {
  View,
  Button,
  TextInput,
  StyleSheet,
  Text
} from 'react-native';

import FirebaseService from '../../../services/FirebaseService';
import firebase from 'react-native-firebase';
import { Actions } from 'react-native-router-flux';

export default class SignUp extends React.Component {
  state = {
    nome_usuario: '', senha: '', email: '', telefone: ''
  }
  onChangeText = (key, val) => {
    this.setState({ [key]: val })
  }
  signUp = async () => {  
    try {
      await firebase.auth()
        .createUserWithEmailAndPassword(this.state.email, this.state.senha)
        .then((res) => {       
          const newid = FirebaseService.pushData('usuarios', this.state);
          alert('Cadastro feito com sucesso!')
          //redirecionar para tela de login
        }).catch(error => this.setState({ errorMessage: error.message }))    
    }    
    catch (err) {
      alert('Erro no cadastro: ', err)
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Cadastro do iTrade</Text>
        <TextInput
          style={styles.input}
          placeholder='Nome'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('nome_usuario', val)}
        />
        <TextInput
          style={styles.input}
          placeholder='Senha'
          secureTextEntry={true}
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('senha', val)}
        />
        <TextInput
          style={styles.input}
          placeholder='Email'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('email', val)}
        />
        <TextInput
          style={styles.input}
          placeholder='Telefone'
          autoCapitalize="none"
          placeholderTextColor='#CD7F32'
          onChangeText={val => this.onChangeText('telefone', val)}
        />
        <Button
          color='#239033'
          title='Registrar'
          onPress={this.signUp}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  input: {
    height: 40,
    backgroundColor: '#E3E1E1',
    marginBottom: 10,
    padding: 10,
    color: '#000'
  },
  container: {
    padding: 20,
    flex: 1,
    backgroundColor: '#FFFAFA',
    alignContent: 'center'
  },
  title: {
    fontSize: 20,
    color: '#CD7F32',
    marginBottom: 30
  }
})
