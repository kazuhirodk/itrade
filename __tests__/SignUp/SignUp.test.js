import SignUp from '../../src/components/SignUp/SignUp.js'
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const wrapper = shallow(<SignUp />);

describe('#SignUp', () => {
  it('matchs snapshot properly', () => {
    const jsonSignUp = renderer.create(<SignUp />).toJSON();
    expect(jsonSignUp).toMatchSnapshot();
  });

  describe('#onChangeText', () => {
    it('receive one or more info and add to state', () => {
      wrapper.instance().onChangeText('name', 'Nome do Usuário');
      wrapper.instance().onChangeText('email', 'email@teste.com');

      expect(wrapper.instance().state.name).toEqual('Nome do Usuário');
      expect(wrapper.instance().state.email).toEqual('email@teste.com');
    })
  });
});
