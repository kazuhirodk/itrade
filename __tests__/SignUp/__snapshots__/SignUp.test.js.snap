// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SignUp matchs snapshot properly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#FFFAFA",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#CD7F32",
        "fontSize": 20,
        "marginBottom": 30,
      }
    }
  >
    Cadastro do iTrade
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Nome"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Senha"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Telefone"
    placeholderTextColor="#CD7F32"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#E3E1E1",
        "color": "#000",
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "height": 50,
        "width": 100,
      }
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Free",
            "textColor": undefined,
            "value": "Free",
          },
          Object {
            "label": "Premium",
            "textColor": undefined,
            "value": "Premium",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#239033",
            },
          ]
        }
      >
        Registrar
      </Text>
    </View>
  </View>
</View>
`;
